src/encoder.rs

```diff
@@ -2322,4 +2369,57 @@ mod test {
     assert_eq!(RAV1E_PARTITION_TYPES[RAV1E_PARTITION_TYPES.len() - 1],
                PartitionType::PARTITION_SPLIT);
   }
-}
+
+  fn create_frame(frame_width: usize, frame_height: usize) -> (Frame<u16>, FrameInvariants<u16>) {
+    let mut frame = Frame::<u16>::new(frame_width, frame_height, ChromaSampling::Cs420);
+
+    // Each pixel contains the sum of its row and column indices:
+    //
+    //  0 1 2 3 4 . .
+    //  1 2 3 4 5 . .
+    //  2 3 4 5 6 . .
+    //  3 4 5 6 7 . .
+    //  4 5 6 7 8 . .
+    //  . . . . . . .
+    //  . . . . . . .
+    for plane in &mut frame.planes {
+      let PlaneConfig { width, height, .. } = plane.cfg;
+      let mut slice = plane.as_mut_slice();
+      for col in 0..width {
+        for row in 0..height {
+          slice[row][col] = (row+col) as u16;
+        }
+      }
+    }
+
+    let config = EncoderConfig {
+      width: frame_width,
+      height: frame_height,
+      quantizer: 100,
+      speed_settings: SpeedSettings::from_preset(10),
+      ..Default::default()
+    };
+    let sequence = Sequence::new(&Default::default());
+    let fi = FrameInvariants::new(config, sequence);
+    (frame, fi)
+  }
+
+  #[test]
+  fn check_activity_mask() {
+    let (frame, fi) = create_frame(64, 64);
+    let mut fs = FrameState::new_with_frame(&fi, Arc::new(frame));
+    fs.compute_activity_mask();
+    let actual_mask = &fs.activity_mask.data;
+    let activity = [
+      65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535,
+      65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535,
+      65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535,
+      65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535,
+      65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535,
+      65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535,
+      65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535,
+      65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535u16];
```

Comment:
This can be simplified to `let activity = [65535u16; 64];` which may be inlined to the following line.